### ISSUE TITLE
fstype xfs not supported

### ISSUE THREAD

**Issue by Besstheone** [wetopi/docker-volume-rbd#23]
Hello,
when creating a new volume with fstype xfs, the volume is not created with the error :
`Create Message=unable to create ceph rbd image: exit status 1`

PS: it is working with ext4

**Comment by sitamet** [comment]
Hi,
What are your config vars?
you should set this two:
```
fstype: optional, defauls to ext4
mkfsOptions: optional, defaults to '-O mmp' (Multiple Mount Protection)
```

**Comment by Atari365** [comment]
I am also not able to create xfs volume. The command is `docker volume create -d wetopi/rbd -o fstype=xfs rbd_vol`. Error is the same: `Error response from daemon: create rbd_vol: VolumeDriver.Create: volume-rbd Name=rbd_vol Request=Create Message=unable to create ceph rbd image: exit status 1`. If I replace xfs with ext4 the volume is created just fine.

**Comment by Atari365** [comment]
Ok, the solution is to set the custom `mkfsOptions` with xfs. There are no `-O` option in `mkfs.xfs`, so it is complains about it.
```
mkfs.xfs: invalid option -- 'O'
unknown option -O
Usage: mkfs.xfs
```

**Comment by sitamet** [comment]
Fixed on release 4.1.0
Moved hard-coded defaults to config vars:

```conf
LOG_LEVEL=[0:ErrorLevel; 1:WarnLevel; 2:InfoLevel; 3:DebugLevel] defaults to 0

RBD_CONF_DEVICE_MAP_ROOT="/dev/rbd"
RBD_CONF_POOL="ssd"
RBD_CONF_CLUSTER=ceph
RBD_CONF_KEYRING_USER=client.admin
RBD_CONF_NAMESPACE=""

Mount options defaults to "--options=noatime" (extended syntax with no spaces)
MOUNT_OPTIONS="--options=noatime"

VOLUME_FSTYPE="ext4"
VOLUME_MKFS_OPTIONS="-O mmp"
VOLUME_SIZE="512"
VOLUME_ORDER="22"
```

now you can install the plugin and set the conf, e.g.

```sh
docker plugin install --grant-all-permissions wetopi/rbd:4.1.0 \
--alias=wetopi/rbd \
LOG_LEVEL=1 \
RBD_CONF_POOL="ssd" \
RBD_CONF_CLUSTER=ceph \
RBD_CONF_KEYRING_USER=client.admin \
VOLUME_ORDER=20 \
VOLUME_MKFS_OPTIONS=""
```

**Comment by GiamBoscaro** [comment]
Hello, I have the same problem creating the xfs image. I have installed the plugin using these settings:

> Fixed on release 4.1.0 Moved hard-coded defaults to config vars:
> 
> ```
> LOG_LEVEL=[0:ErrorLevel; 1:WarnLevel; 2:InfoLevel; 3:DebugLevel] defaults to 0
> 
> RBD_CONF_DEVICE_MAP_ROOT="/dev/rbd"
> RBD_CONF_POOL="ssd"
> RBD_CONF_CLUSTER=ceph
> RBD_CONF_KEYRING_USER=client.admin
> RBD_CONF_NAMESPACE=""
> 
> Mount options defaults to "--options=noatime" (extended syntax with no spaces)
> MOUNT_OPTIONS="--options=noatime"
> 
> VOLUME_FSTYPE="ext4"
> VOLUME_MKFS_OPTIONS="-O mmp"
> VOLUME_SIZE="512"
> VOLUME_ORDER="22"
> ```
> 
> now you can install the plugin and set the conf, e.g.
> 
> ```shell
> docker plugin install --grant-all-permissions wetopi/rbd:4.1.0 \
> --alias=wetopi/rbd \
> LOG_LEVEL=1 \
> RBD_CONF_POOL="ssd" \
> RBD_CONF_CLUSTER=ceph \
> RBD_CONF_KEYRING_USER=client.admin \
> VOLUME_ORDER=20 \
> VOLUME_MKFS_OPTIONS=""
> ```

Hello, I have the same problem creating the xfs image. I have installed the plugin using your settings, then I try to create the volume with:

`docker volume create -d wetopi/rbd -o size=1024 -o fstype=xfs my_rbd_volume`

but I get:

`Error response from daemon: create my_rbd_volume: VolumeDriver.Create: volume-rbd Name=my_rbd_volume Request=Create Message=unable to create ceph rbd image: exit status 1`

What am I doing wrong?

**Comment by GiamBoscaro** [comment]
> Hello, I have the same problem creating the xfs image. I have installed the plugin using these settings:
> 
> > Fixed on release 4.1.0 Moved hard-coded defaults to config vars:
> > ```
> > LOG_LEVEL=[0:ErrorLevel; 1:WarnLevel; 2:InfoLevel; 3:DebugLevel] defaults to 0
> > 
> > RBD_CONF_DEVICE_MAP_ROOT="/dev/rbd"
> > RBD_CONF_POOL="ssd"
> > RBD_CONF_CLUSTER=ceph
> > RBD_CONF_KEYRING_USER=client.admin
> > RBD_CONF_NAMESPACE=""
> > 
> > Mount options defaults to "--options=noatime" (extended syntax with no spaces)
> > MOUNT_OPTIONS="--options=noatime"
> > 
> > VOLUME_FSTYPE="ext4"
> > VOLUME_MKFS_OPTIONS="-O mmp"
> > VOLUME_SIZE="512"
> > VOLUME_ORDER="22"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > now you can install the plugin and set the conf, e.g.
> > ```shell
> > docker plugin install --grant-all-permissions wetopi/rbd:4.1.0 \
> > --alias=wetopi/rbd \
> > LOG_LEVEL=1 \
> > RBD_CONF_POOL="ssd" \
> > RBD_CONF_CLUSTER=ceph \
> > RBD_CONF_KEYRING_USER=client.admin \
> > VOLUME_ORDER=20 \
> > VOLUME_MKFS_OPTIONS=""
> > ```
> 
> Hello, I have the same problem creating the xfs image. I have installed the plugin using your settings, then I try to create the volume with:
> 
> `docker volume create -d wetopi/rbd -o size=1024 -o fstype=xfs my_rbd_volume`
> 
> but I get:
> 
> `Error response from daemon: create my_rbd_volume: VolumeDriver.Create: volume-rbd Name=my_rbd_volume Request=Create Message=unable to create ceph rbd image: exit status 1`
> 
> What am I doing wrong?

Solved by adding a random options even if I didn't need one, like:

`docker volume create -d wetopi/rbd -o size=1024 -o fstype=xfs -o mkfsOptions="-L my_label" my_rbd_volume`
